### ISSUE TITLE
Long-lived requests can starve Tap/Top clients

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#1729]
When tapping a service like the LInkerd controller, tap & top will run for a while but can easily become "frozen" -- no new events are observed by the client. I noticed that there was a single request remaining in tap's index. The tap client sees the events:

```
req id=0:228 proxy=in  src=10.24.0.31:60614 dst=10.24.0.30:8086 tls=disabled :method=POST :authority=proxy-api.linkerd.svc.cluster.local:8086 :path=/io.linkerd.proxy.destination.Destination/Get
rsp id=0:228 proxy=in  src=10.24.0.31:60614 dst=10.24.0.30:8086 tls=disabled :status=200 latency=3199µs
```

The `end` message for such a request may be received some arbitrarily long time later.

Because the tap client does not initiate a new tap request until all of its requests have fully completed, it sits idle indefinitely (until the end event is received).

**Comment by olix0r** [comment]
This also means that Top will basically never display summaries for these long-lived requests, as if they don't even exist.

For instance, when tapping the controller pod, we never see Destination/Get requests in the results.

We will very rarely see tap requests:

```
controller-589bcb468c-9n10.24.0.31              POST       /io.linkerd.proxy.tap.Tap/Observe     2      17s    18s    18s    100.00%
```

but i suspect that we d not represent these requests proportionally.

**Comment by siggy** [comment]
Fixed via linkerd/linkerd2-proxy#299
